packages/aws-cdk/lib/init.ts

```diff
@@ -156,6 +162,22 @@ export class InitTemplate {
     }
   }
 
+  private needsShell(language: string, type: string): boolean {
+    return (language === 'csharp' || language === 'fsharp') && (type === 'app' || type === 'default');
```

Comment:
This scares me. Duplicate bookkeeping of this kind of information feels a tad brittle.

Reply:
Gone